### PR TITLE
Add a setting for volume envelope processing

### LINKED ISSUE #70: Add a setting for volume envelope processing (compliant or EMU)
Starting with FluidSynth 1.0.9, the volume envelope was changed to be compatible with the EMU10K1. I propose that a setting is added, such as synth.volenv with the options "compliant" or "emu", with "emu" as the default. "compliant" would use the previous behavior which is compliant with the SoundFont specification. This ticket was created in response to a message from Sebastian Biallas:  

<http://lists.gnu.org/archive/html/fluid-dev/2010-01/msg00011.html>  

Reported by: elementgreen

Original Ticket: [fluidsynth/tickets/70](https://sourceforge.net/p/fluidsynth/tickets/70)

### PR DESCRIPTION
**Pull request by derselbst**
fix #70

### CODE REVIEW

**Comment by derselbst** [comment]
@ensonic Could you please read through the doc of `synth.volenv`. Is it clear enough?

File: doc/fluidsynth.1 [hunk 1]
```diff
@@ -221,6 +221,9 @@ Must always to be true for usage by fluidsynth executable.
 .TP
 .B synth.verbose            BOOL  [def=False]
 Print received MIDI events to stdout.
+.TP
+.B synth.volenv             STR   [def='emu' vals:'compliant', 'emu']
```
Comment (ensonic) [inline doc/fluidsynth.1]: Would it make sense to have the previous behaviour as the default (compliant)?

File: src/synth/fluid_synth.c [hunk 2]
```diff
@@ -554,8 +557,29 @@ new_fluid_synth(fluid_settings_t *settings)
...
+    {
+            fluid_conversion_set_atten_power(FLUID_ATTEN_POWER_DEFAULT_COMPLIANT);
+    }
+    else if (fluid_settings_str_equal (settings, "synth.volenv", "emu") == 1)
+    {
+            fluid_conversion_set_atten_power(FLUID_ATTEN_POWER_DEFAULT_EMU);
+    }
+    else
+    {
+        if (fluid_settings_copystr(settings, "synth.volenv", buf, 64) == 1)
```
Comment (ensonic) [inline src/synth/fluid_synth.c]: if (fluid_settings_copystr(settings, "synth.volenv", buf, sizeof(buf)) == 1)

File: src/utils/fluid_conv.c [hunk 3]
```diff
@@ -21,6 +21,12 @@
 #include "fluid_conv.h"
 
 
+/* EMU 8k/10k don't follow spec in regards to volume attenuation.
```
Comment (ensonic) [inline src/utils/fluid_conv.c]: doesn't follow

File: src/utils/fluid_conv.c [hunk 4]
```diff
@@ -21,6 +21,12 @@
...
+ * This factor is used in the equation pow (10.0, cb / FLUID_ATTEN_POWER_FACTOR).
+ * By the standard this should be -200.0. */
+/* 07/11/2008 modified by S. Christian Collins for increased velocity sensitivity.  Now it equals the response of EMU10K1 programming.*/
```
Comment (ensonic) [inline src/utils/fluid_conv.c]: break long line?

File: doc/fluidsynth.1 (hunk 1 above)
Comment (derselbst) [reply doc/fluidsynth.1]: Well... I'm not sure. [Element suggested "emu" to be the default](https://github.com/FluidSynth/fluidsynth/issues/70#issue-238323938) and I'm also not feeling comfortable changing this again to the previous behaviour. So I think we should leave it as it is.

File: src/synth/fluid_synth.c (hunk 2 above)
Comment (derselbst) [reply src/synth/fluid_synth.c]: you're right

File: src/utils/fluid_conv.c (hunk 3 above)
Comment (derselbst) [reply src/utils/fluid_conv.c]: I consider 8k/10k to be plural, sry :)

File: src/utils/fluid_conv.c (hunk 4 above)
Comment (derselbst) [reply src/utils/fluid_conv.c]: clang-format (#206) should take care of that some when in the future.

**Pull request merged**